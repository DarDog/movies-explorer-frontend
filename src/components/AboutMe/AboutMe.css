.about-me {
  padding: 110px 70px 0;
  background-color: white;
}

.about-me__info {
  display: flex;
  justify-content: space-between;
}

.about-me__container {
  width: 100%;
  padding-right: 50px;
}

.about-me__name {
  margin: 0 0 20px;
  font-weight: 400;
  font-size: 50px;
  line-height: 58px;
  letter-spacing: -0.04em;
  color: black;
}

.about-me__job {
  margin: 0 0 26px;
  font-weight: 500;
  font-size: 18px;
  line-height: 20px;
  color: black;
}

.about-me__paragraph {
  max-width: 600px;
  margin: 0 0 100px;
  font-weight: 400;
  font-size: 14px;
  line-height: 22px;
  color: black;
}

.about-me__link {
  text-decoration: none;
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  color: black;
}

.about-me__item {
  margin-right: 20px;
  transition: .3s;
}

.about-me__item:hover {
  transform: scale(1.5);
}

.about-me__image {
  max-width: 270px;
  object-fit: cover;
  object-position: 65%;
  border-radius: 10px;
}

@media screen and (max-width: 768px){
  .about-me {
    padding: 90px 50px 0;
  }

  .about-me__container {
    max-width: 364px;
  }

  .about-me__name {
    font-size: 40px;
    line-height: 40px;
  }

  .about-me__job {
    font-size: 12px;
    line-height: 18px;
  }

  .about-me__paragraph {
    font-size: 12px;
    line-height: 18px;
  }

  .about-me__image {
    max-width: 255px;
    max-height: 307px;
  }
}
