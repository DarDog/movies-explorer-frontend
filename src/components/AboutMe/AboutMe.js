import React from 'react';
import SectionTitle from "../SectionTitle/SectionTitle";
import Ul from "../Ul/Ul";
import './AboutMe.css'

const AboutMe = () => {
  return (
    <section className='about-me' id='about-me'>
      <SectionTitle>Студент</SectionTitle>
      <div className='about-me__info'>
        <div className='about-me__container'>
          <h3 className='about-me__name'>Владислав</h3>
          <p className='about-me__job'>Фронтенд-разработчик, 24 года</p>
          <p className='about-me__paragraph'>Я родом из города Южно-Сахалинск. Готов как к работе удаленно в любом
            часовом поясе так и к переезду.
            Увлекаюсь фотографией и настольными играми. Я 3 года работал оператором 1С в розничной компании, но решил
            сменить деятельность и занялся веб-разработкой. Сейчас я ищу свою работу мечты и возможно я ваш сотрудник
            мечты.</p>
          <Ul>
            <li className='about-me__item'>
              <a href="https://vk.com/me_here_fine" className='about-me__link'>VK</a>
            </li>
            <li className='about-me__item'>
              <a href="https://github.com/DarDog" className='about-me__link'>Github</a>
            </li>
          </Ul>
        </div>
        <img
          className='about-me__image'
          src="https://sun9-79.userapi.com/impg/UNNsPwvOp3BZIil4jO23M9rSxpjC5hBjb3DqTw/2RerGfe7TJo.jpg?size=1400x934&quality=96&sign=b4e0c3a162765733fa367a2291d1926d&type=album"
          alt="Моя фотография"
        />
      </div>
    </section>
  );
}

export default AboutMe;
