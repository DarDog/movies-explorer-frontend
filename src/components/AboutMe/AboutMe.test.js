import React from 'react';
import AboutMe from "./AboutMe";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';


describe('Компонент Tech', () => {
  it('Рендерится', () => {
    render(<AboutMe/>);

    expect(screen.getByRole('heading', {level: 2})).toBeInTheDocument();
    expect(screen.getByText('Студент')).toBeInTheDocument();
    expect(screen.getByRole('heading', {level: 3})).toBeInTheDocument();
    expect(screen.getByText('Владислав')).toBeInTheDocument();
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByRole('list')).toBeInTheDocument();
  })
  it('Соответствует своему snapshot', () => {
    const component = render(<AboutMe/>);

    expect(component).toMatchSnapshot();
  })
})
