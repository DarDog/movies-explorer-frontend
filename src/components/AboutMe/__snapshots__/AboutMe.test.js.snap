// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Tech Соответствует своему snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="about-me"
      >
        <h2
          class="title"
        >
          Студент
        </h2>
        <div
          class="about-me__info"
        >
          <div>
            <h3
              class="about-me__name"
            >
              Владислав
            </h3>
            <p
              class="about-me__job"
            >
              Фронтенд-разработчик, 24 года
            </p>
            <p
              class="about-me__paragraph"
            >
              Я родом из города Южно-Сахалинск. Готов как к работе удаленно в любом часовом поясе так и к переезду. Увлекаюсь фотографией и настольными играми. Я 3 года работал оператором 1С в розничной компании, но решил сменить деятельность и занялся веб-разработкой. Сейчас я ищу свою работу мечты и возможно я ваш сотрудник мечты.
            </p>
            <ul
              class="list  "
            >
              <li
                class="about-me__item"
              >
                <a
                  class="about-me__link"
                  href="https://vk.com/me_here_fine"
                >
                  VK
                </a>
              </li>
              <li
                class="about-me__item"
              >
                <a
                  class="about-me__link"
                  href="https://github.com/DarDog"
                >
                  Github
                </a>
              </li>
            </ul>
          </div>
          <img
            alt="Моя фотография"
            class="about-me__image"
            src="https://sun9-79.userapi.com/impg/UNNsPwvOp3BZIil4jO23M9rSxpjC5hBjb3DqTw/2RerGfe7TJo.jpg?size=1400x934&quality=96&sign=b4e0c3a162765733fa367a2291d1926d&type=album"
          />
        </div>
        <p
          class="about-me__portfolio"
        >
          Портфолио
        </p>
        <ul
          class="list portfolio__list list_type_column"
        >
          <li
            class="portfolio__item"
          >
            <a
              class="portfolio__link"
              href="https://dardog.github.io/how-to-learn/"
            >
              Статичный сайт
            </a>
            <div
              class="link__arrow"
            />
          </li>
          <li
            class="portfolio__item"
          >
            <a
              class="portfolio__link"
              href="https://dardog.github.io/russian-travel/"
            >
              Адаптивный сайт
            </a>
            <div
              class="link__arrow"
            />
          </li>
          <li
            class="portfolio__item"
          >
            <a
              class="portfolio__link"
              href="http://mesto.subb.nomoredomains.rocks/sign-in"
            >
              Одностраничное приложение
            </a>
            <div
              class="link__arrow"
            />
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="about-me"
    >
      <h2
        class="title"
      >
        Студент
      </h2>
      <div
        class="about-me__info"
      >
        <div>
          <h3
            class="about-me__name"
          >
            Владислав
          </h3>
          <p
            class="about-me__job"
          >
            Фронтенд-разработчик, 24 года
          </p>
          <p
            class="about-me__paragraph"
          >
            Я родом из города Южно-Сахалинск. Готов как к работе удаленно в любом часовом поясе так и к переезду. Увлекаюсь фотографией и настольными играми. Я 3 года работал оператором 1С в розничной компании, но решил сменить деятельность и занялся веб-разработкой. Сейчас я ищу свою работу мечты и возможно я ваш сотрудник мечты.
          </p>
          <ul
            class="list  "
          >
            <li
              class="about-me__item"
            >
              <a
                class="about-me__link"
                href="https://vk.com/me_here_fine"
              >
                VK
              </a>
            </li>
            <li
              class="about-me__item"
            >
              <a
                class="about-me__link"
                href="https://github.com/DarDog"
              >
                Github
              </a>
            </li>
          </ul>
        </div>
        <img
          alt="Моя фотография"
          class="about-me__image"
          src="https://sun9-79.userapi.com/impg/UNNsPwvOp3BZIil4jO23M9rSxpjC5hBjb3DqTw/2RerGfe7TJo.jpg?size=1400x934&quality=96&sign=b4e0c3a162765733fa367a2291d1926d&type=album"
        />
      </div>
      <p
        class="about-me__portfolio"
      >
        Портфолио
      </p>
      <ul
        class="list portfolio__list list_type_column"
      >
        <li
          class="portfolio__item"
        >
          <a
            class="portfolio__link"
            href="https://dardog.github.io/how-to-learn/"
          >
            Статичный сайт
          </a>
          <div
            class="link__arrow"
          />
        </li>
        <li
          class="portfolio__item"
        >
          <a
            class="portfolio__link"
            href="https://dardog.github.io/russian-travel/"
          >
            Адаптивный сайт
          </a>
          <div
            class="link__arrow"
          />
        </li>
        <li
          class="portfolio__item"
        >
          <a
            class="portfolio__link"
            href="http://mesto.subb.nomoredomains.rocks/sign-in"
          >
            Одностраничное приложение
          </a>
          <div
            class="link__arrow"
          />
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
